lib/rubocop/cop/rspec/unspecified_exception.rb

```diff
@@ -0,0 +1,83 @@
+module RuboCop
+  module Cop
+    module RSpec
+      # Checks for a specified error in checking raised errors.
+      #
+      # Enforces one of an Exception type, a string, or a regular
+      # expression to match against the exception message as a parameter
+      # to `raise_error`
+      #
+      # @example
+      #
+      #     # bad
+      #     expect {
+      #       raise StandardError.new('error')
+      #     }.to raise_error
+      #
+      #     # good
+      #     expect {
+      #       raise StandardError.new('error')
+      #     }.to raise_error(StandardError)
+      #
+      #     expect {
+      #       raise StandardError.new('error')
+      #     }.to raise_error(StandardError) { |err| expect(err.data).to eq(42) }
+      #
+      #     expect {
+      #       raise StandardError.new('error')
+      #     }.to raise_error(StandardError) do |err|
+      #       expect(err.data).to eq(42)
+      #     end
```

Comment:
the difference is the style of block (curly braces vs. `do`/`end` and while I was parsing the AST one passed while the other did not.  It's been about three months since I did that work so I don't remember the exact details of the issue but I suspect it was because of the different bindings of the two types of blocks.  Is that reasonable?

Reply:
I'm not sure we need so much good examples in the documentation, as most of them include `to raise_error(StandardError)`. It's good that you enumerated all of them for the tests though

Reply:
That's fair, any feelings on which particular ones I should remove?

Reply:
I would leave the first one (that contains only `raise_error(StandardError)`) as it's shortest and cover the fact that you have to pass an error class

Reply:
cool.  Done!

Reply:
You could leave the ones that use the message though :)